src/home/getters/get-github-user.ts

```diff
@@ -30,13 +27,33 @@ async function getNewSessionToken(): Promise<string | null> {
   const params = new URLSearchParams(hash.substr(1)); // remove the '#' and parse
   const providerToken = params.get("provider_token");
   if (!providerToken) {
-    return null;
+    const code = params.get("error_code");
+
+    // supabase auth provider has failed for some reason
+    if (code === "500") {
+      displayPopupMessage(`GitHub Login Provider`, `Your access token may have reached its rate limit, please try again after one hour.`);
+      console.error("GitHub login provider");
+    }
   }
-  return providerToken;
+
+  return providerToken || null;
 }
 
-async function getNewGitHubUser(providerToken: string): Promise<GitHubUser> {
+async function getNewGitHubUser(providerToken: string | null): Promise<GitHubUser | null> {
   const octokit = new Octokit({ auth: providerToken });
-  const response = (await octokit.request("GET /user")) as GitHubUserResponse;
-  return response.data;
+  try {
+    const response = (await octokit.request("GET /user")) as GitHubUserResponse;
+    return response.data;
+  } catch (error: unknown) {
+    if (error.status === 403 && error.message.includes("API rate limit exceeded")) {
```

Comment:
Instead of checking the error message string, I think it might be wiser to call https://octokit.github.io/rest.js/v20#rate-limit which seems to be there for this purpose?

If it works fine, the same principle can be applied on line 34.

Reply:
This only works for the line you've suggested here as in line 34 we do not have an `auth-token` as the `auth provider` (supabase gh integration) has returned an error string instead of the user's token, at least in the specific case I as covering.

I can use that endpoint in both but default to a hardcoded `1 hour` if we don't have a token in L34

Reply:
realized that we have access to the header response regardless of `auth-token` and the call flow allows us to use those headers, so I've removed the hard `1 hour` outcome